modules/references/client/components/ReferencesNew.component.js

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as references from './references.api';
+import Navigation from './Navigation';
+import Interaction from './Interaction';
+import Recommend from './Recommend';
+import { Self, Loading, Duplicate, Submitted } from './Info';
+import { Tab, Tabs } from 'react-bootstrap';
+
+const api = { references };
+
+export default class ReferencesNew extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      tab: 0,
+      reference: {
+        interactions: {
+          met: false,
+          hostedThem: false,
+          hostedMe: false
+        },
+        recommend: null
+      },
+      report: false,
+      reportMessage: '',
+      isSelf: props.userFrom._id === props.userTo._id,
+      isLoading: true,
+      isSubmitting: false,
+      isDuplicate: false,
+      isSubmitted: false,
+      isPublic: false
+    };
+  }
+
+  async componentDidMount() {
+    const reference = await api.references.read({ userFrom: this.props.userFrom._id, userTo: this.props.userTo._id });
+
+    const newState = { isLoading: false };
+
+    if (reference.length === 1) newState.isDuplicate = true;
+
+    this.setState(newState);
+  }
+
+  handleTabSwitch(move) {
+    this.setState({
+      tab: this.state.tab + move
+    });
+  }
+
+  handleChangeInteraction(interactionType) {
+    const interaction = { };
+    interaction[interactionType] = !this.state.reference.interactions[interactionType];
+    this.setState({
+      reference: {
+        ...this.state.reference,
+        interactions: {
+          ...this.state.reference.interactions,
+          ...interaction
+        }
+      }
+    });
+  }
```

Comment:
This `handleChangeInteraction` seems pretty complicated and hard to understand what's going on here.

Its toggling one of the values in `interactions`, right?
```
      reference: {
        interactions: {
          met: false,
          hostedThem: false,
          hostedMe: false
        },
        recommend: null
      },
```

...which generally seems like a great use case for [Lodash `update`](https://lodash.com/docs/4.17.11#update).

That said, [nested states are not very React'y](https://stackoverflow.com/a/51136076) and don't provide great performance.

I know we've just inherited them from Angular but ideally, these would just be:
```js
state = {
  interractionMet: false,
  interractionhostedThem: false,
  interractionhostedMe: false,
  recommend: true,
}
```

I think this is already showing signs that we'll need some sort of data layer to handle all this.

At the current state (pun intended) of the application, it'll be messy to deal with using manually building back the object for API calls because we want the data being sent to API in the same format to API as we store it in the DB.

We'd want to store the data in the same format in client side as well, and for that, we'll need a data layer. Components, on the other hand, shouldn't deal with mongoose document structures — they should be super simple and just care about their own state and props.

All this is to say that we'll probably make our future life easier by:
- keeping component state shallow and simple
- manipulating the state back to correct object before sending it to API

Reply:
4b9a8a585ece7cec53cd399ea79a41a7967bb617 shallowing the component state and manipulating the state from and to API format.